client/src/game/reducers/cardDeductionReducer.ts

```diff
@@ -18,23 +18,23 @@ const cardDeductionReducer = (
     case 'discard':
     case 'play':
     case 'draw': {
-      return makeDeductions(current(hands), deck, variant, metadata);
+      return makeDeductions(deck, current(createDraft(hands)), metadata);
```

Comment:
don't use createDraft here. Are you passing hands that are not draft to it? Maybe a better solution is to use current() on the caller (cardsReducer)

Reply:
I don't really understand immer, so that doesn't help :).  My tests in `cardsReducer.test.ts` were passing hands that weren't draft.  I tried to do `createDraft` under `initialGameState` but that didn't work.  I think I was maybe missing tests that overwrote hands.  That said, maybe the tests shouldn't be using `createDraft` either and should be doing something different to get the `current` call to work.

Reply:
The right solution is to ensure the call to current happens at cardsReducer (the caller to this). Then your tests don't have to worry. You should never do createDraft.

Reply:
The tests for cardsReducer still fail (complaining that it's not a draft).  If I move it up further to gameStateReducer then it still complains in the integration test.  I can probably just mock the `current` method in my tests and that wouldn't be a very big change.

I should note that I'm testing `cardDeductionReducer` via `cardsReducer.test` (kind of treating them as one unit).  If you'd like, I can mock `cardDeductionReducer` in `cardsReducer.test` and make a separate `cardDeductionReducer.test`.  It still wouldn't solve the `current` issue though because I believe the remaining tests would fail in `cardsReducer.test` (or `integration.test`).

Is `current` supposed to speed the code up?  The [docs](https://immerjs.github.io/immer/docs/current) for `current` mainly seem to say that it's useful to call that method for debugging?  It's taking a (mutable) draft and creating another (mutable) copy?  You probably don't need to delve too deep into the details here... I just don't get the impression that `current` is supposed to improve performance when reading the docs.

Reply:
I think there's some misuse of drafting. The perf benefit is that reading from drafts is REALLY slow (they are proxies). Current converts a draft into the real object.

Reply:
You can solve it definitely by having whatever reducer that changes hands into a draft (likely cardsReducer) call current when it's done.

Reply:
I think `gameStateReducer` creates a draft of `GameState`, which `cardsReducer` has passed in as a parameter.  I think I was putting `current` around the wrong thing when I previously tried to put it in `gameStateReducer`.  I'm not sure if we want to copy the whole `GameState` out of draft when calling `cardsReducer` or not.  I put a `current` call in `cardsReducer` when calling `game.hands` and then mocked out the `current` call in `cardsReducer.test` and that seems to work.